### PR TITLE
Introduce revision 2 of the IRQ_STEER driver

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
Tested locally using Zephyr and i.MX8MP and i.MX8QM.

### CODE REVIEW

**Comment by LaurentiuM1234** [comment]
@mcuxsusan @zejiang0jason pls review.

File: drivers/irqsteer/fsl_irqsteer_rev2.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#define FSL_COMPONENT_ID "platform.drivers.irqsteer"
+#endif
+
+/*******************************************************************************
+ * Code
+ ******************************************************************************/
+
+uint32_t IRQSTEER_GetMasterIrqCount(IRQSTEER_Type *base, irqsteer_int_master_t intMasterIndex)
+{
+    if (!(FSL_FEATURE_IRQSTEER_IRQ_NUM % 64)) {
```
Comment (zejiang0jason) [inline drivers/irqsteer/fsl_irqsteer_rev2.c:26]: Is it better to use `FSL_FEATURE_IRQSTEER_CHN_MASK_COUNT % 2` ? I am thinking that whether it is possible that IRQSTEER slot is not fully used, for example, there are 4 groups, but only (4 * 32 - 1) interrupt sources.

File: drivers/irqsteer/fsl_irqsteer_rev2.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/* Component ID definition, used by tools. */
+#ifndef FSL_COMPONENT_ID
+#define FSL_COMPONENT_ID "platform.drivers.irqsteer"
+#endif
+
+/*******************************************************************************
+ * Code
+ ******************************************************************************/
+
+uint32_t IRQSTEER_GetMasterIrqCount(IRQSTEER_Type *base, irqsteer_int_master_t intMasterIndex)
```
Comment (zejiang0jason) [inline drivers/irqsteer/fsl_irqsteer_rev2.c:24]: Just confirm, can this function be static? thanks.

Comment (LaurentiuM1234) [reply drivers/irqsteer/fsl_irqsteer_rev2.c:24]: Can't make it static as it's currently being used by the Zephyr driver

File: drivers/irqsteer/fsl_irqsteer_rev2.c (hunk 1 above)
Comment (LaurentiuM1234) [reply drivers/irqsteer/fsl_irqsteer_rev2.c:26]: How would you detect such an anomaly using just `FSL_FEATURE_IRQSTEER_CHN_MASK_COUNT`? At the moment, `IRQSTEER_GetMasterIrqCount` is rather simplistic and it's not really meant to handle such anomalies. In the future, if need be we can add support for such cases but for now I would suggest leaving it as-is as I don't have an example of such an anomaly in mind (for i.MX8QM and i.MX8MP this doesn't happen as far as I can tell). Do you have a specific platform in mind?

**Pull request closed without merging**